include/vast/Interfaces/ClangAST/ASTDeclInterface.td

```diff
@@ -0,0 +1,129 @@
+    [ASTNamedDeclInterface] > {
+
+    let description = [{
+        Interface for clang's ValueDecl AST node.
+    }];
+
+    let methods = [
+        InterfaceMethod< "", "void", "getType", (ins) >
+    ];
+}
+
+def ASTDeclaratorDeclInterface : VastOpInterface< "ASTDeclaratorDeclInterface",
+    [ASTValueDeclInterface] > {
+    
+    let description = [{
+        Interface for clang's DeclaratorDecl AST node.
+    }];
+
+    let methods = [];
+}
+
+def ASTFieldDeclInterface : VastOpInterface< "ASTFieldDeclInterface",
+    [ASTDeclaratorDeclInterface] > {
+    let description = [{
+        Interface for clang's FieldDecl AST node.
+    }];
+
+    let methods = [
+        InterfaceMethod< "Determines whether this is an unnamed bitfield",
+        "bool", "isUnnamedBitField", (ins) >,
+
+        InterfaceMethod< [{ Determine if this field is a subobject of zero size,
+                            that is, either a zero-lenght bit-field or a field
+                            of empty class type with the [[no_unique_address]]
+                            attribute. }], "bool", "isZeroSize", (ins) >,
+
+        InterfaceMethod< "Return ASTContext.", "void", "getASTContext", (ins) >,
+    ];
+}
+
+def ASTVarDeclInterface : VastOpInterface< "ASTVarDeclInterface",
+    [ASTDeclaratorDeclInterface] > {
+    let description = [{
+        Interface for clang's VarDecl AST node.
+    }];
+
+    let methods = [
+        InterfaceMethod< [{ Returns true for local variable declarations other than
+                            parameters. Note that this includes static variables 
+                            inside of a functions. It also includes variables inside
+                            blocks.
+
+                            void foo() { int x; static int y, extern int z; } }],
+                         "bool", "isLocalVarDecl", (ins) >,
+
+        InterfaceMethod< [{ Returns true for all variables that do not have local
+                            storage. This includes all global variables as well as
+                            static variables declared within a fuctnion }], "bool",
+                         "hasGlobalStorage", (ins) >,
+
+        InterfaceMethod< [{ Determine whether this variable is the exception
+                            variable in a C++ catch statement or an Objectinve-C
+                            \@catch statement }], "bool", "isExceptionVariable", (ins) >,
+
+        InterfaceMethod< [{ Whether this variabla is the implicit variable for
+                            a lambda init-capture. }], "bool", "isInitCapture", (ins) >,
+
+        InterfaceMethod< "Should return Expr *", "void", "getInit", (ins) >
+    ];
+}
+
+def ASTRecordDeclInterface : VastOpInterface< "ASTRecordDeclInterface",
+    [ASTTagDeclInterface] > {
+    let description = [{
+       Interface for clang's RecordDecl AST node.
+    }];
+
+    let methods = [
+        InterfaceMethod< "<description>", "std::vector< vast::ASTFieldDeclInterface >",
```

Comment:
This should be some `filed_range` not a vector.

Reply:
We will need something like `filed_range = llvm::iterator_range< specific_decl_iterator< vast::ASTFieldDeclInterface > >`.

This might be more complex so create it an issue and lets resolve it later. Keep vector for now.